x-pack/plugins/dashboard_enhanced/public/services/drilldowns/dashboard_to_dashboard_drilldown/components/collect_config_container.tsx

```diff
@@ -0,0 +1,172 @@
+  isLoading: boolean;
+  selectedDashboard?: EuiComboBoxOptionOption<string>;
+  error?: string;
+}
+
+export class CollectConfigContainer extends React.Component<
+  DashboardDrilldownCollectConfigProps,
+  CollectConfigContainerState
+> {
+  private isMounted = true;
+  state = {
+    dashboards: [],
+    isLoading: false,
+    searchString: undefined,
+    selectedDashboard: undefined,
+    error: undefined,
+  };
+
+  constructor(props: DashboardDrilldownCollectConfigProps) {
+    super(props);
+    this.debouncedLoadDashboards = debounce(this.loadDashboards.bind(this), 500);
+  }
+
+  componentDidMount() {
+    this.loadSelectedDashboard();
+    this.loadDashboards();
+  }
+
+  componentWillUnmount() {
+    this.isMounted = false;
+  }
+
+  render() {
+    const { config, onConfig } = this.props;
+    const { dashboards, selectedDashboard, isLoading, error } = this.state;
+
+    return (
+      <DashboardDrilldownConfig
+        activeDashboardId={config.dashboardId}
+        dashboards={mergeDashboards(dashboards, selectedDashboard)}
+        currentFilters={config.useCurrentFilters}
+        keepRange={config.useCurrentDateRange}
+        isLoading={isLoading}
+        error={error}
+        onDashboardSelect={dashboardId => {
+          onConfig({ ...config, dashboardId });
+          if (this.state.error) {
+            this.setState({ error: undefined });
+          }
+        }}
+        onSearchChange={this.debouncedLoadDashboards}
+        onCurrentFiltersToggle={() =>
+          onConfig({
+            ...config,
+            useCurrentFilters: !config.useCurrentFilters,
+          })
+        }
+        onKeepRangeToggle={() =>
+          onConfig({
+            ...config,
+            useCurrentDateRange: !config.useCurrentDateRange,
+          })
+        }
+      />
+    );
+  }
+
+  private async loadSelectedDashboard() {
+    const {
+      config,
+      params: { start },
+    } = this.props;
+    if (!config.dashboardId) return;
+    const savedObject = await start().core.savedObjects.client.get<{ title: string }>(
+      'dashboard',
+      config.dashboardId
+    );
+
+    if (!this.isMounted) return;
+
+    // handle case when destination dashboard is no longer exist
+    if (savedObject.error?.statusCode === 404) {
+      this.setState({
+        error: txtDestinationDashboardNotFound(config.dashboardId),
+      });
+      this.props.onConfig({ ...config, dashboardId: undefined });
+      return;
+    }
+
+    if (savedObject.error) {
+      this.setState({
+        error: savedObject.error.message,
+      });
+      this.props.onConfig({ ...config, dashboardId: undefined });
+      return;
+    }
+
+    this.setState({ selectedDashboard: dashboardSavedObjectToMenuItem(savedObject) });
+  }
+
+  private readonly debouncedLoadDashboards: (searchString?: string) => void;
+  private async loadDashboards(searchString?: string) {
+    const currentDashboardId = this.props.context.placeContext.embeddable?.parent?.id;
```

Comment:
Unfortunately this is not necessarily the dashboard saved object id, it just happens to be.  An embeddable can be inside a dashboard container that is not backed by a saved object at all, so this id can be anything.

This is also, I think, the only reason this is tied specifically to dashboards as the origin vs, say, canvas.

I see this `PlaceContext` is taken from the `Create drilldown` action - `placeContext={context}` and that context is `EmbeddableContext`.  @ThomThomson is working on making these actions more generic based on origin as well and I think he is going to be passing in origin URL through this context.  Is that still the plan? (I'll sync up with you two off line too bc comments in a big PR like this are hard to find).

But that may be a way to generalize this.. check the URL, see if it's a dashboard app url, extract the id if the origin is on a dashboard.  I think also, to support background search feature on an embeddable level, we'll need to pass down serializeable origin url which means url generator id + state which would make that extraction much easier.

A drilldown to canvas option could do the same thing in order to filter out any current canvas that is being used.

Another option - just show all the ids in the list. Filtering it out doesn't seem that important to me, and it also seems to be the only reason to have this `PlaceContext` generic at all, and I think there are some issues with that. I can write a drilldown passing it any generic `PlaceContext` I want, but that doesn't mean that data will actually be accessible to me at all.  It'll look type safe, but it'll fail at runtime.

Reply:
If we make a `DashboardContainerEmbeddable` "by reference", the input would extend `SavedObjectEmbeddableInput` and you could access it via `input.savedObjectId`.  Right now though it's not by ref, it's by val, and grabbing it from `input.id` only works accidentally because the dashboard app just happens to use it for `id`.  Anyone could put anything in `id` though. It's really only used to find a child in a parent by id, and I have half a mind to remove it entirely, it doesn't even make any sense that an embeddable not inside a container even needs an `id` parameter.

Reply:
I don't see how this is a problem as ID can still stay as `input.id` instead of `input.savedObjectId`; or the drilldown can check both fields.

But for purposes of this PR I've removed `.placeContext` as agreed at our Thu sync, so that line is gone now.